syntax = "proto3";
package oneflow;

import "graph/exec_graph.proto";

enum TaskType {
  HostCompTask = 0;
  DeviceCompTask = 1;
  CopyHdTask = 2;
  CommNetTask = 3;
  BoxingTask = 4;
}

message TaskProto {
  // common
  int64 id = 1;
  TaskType type = 2;
  int64 machine_id = 3;
  int64 thrd_local_id = 4;
  bool is_forward = 5;
  ExecGraphProto exec_graph = 6;
  repeated int64 produced_regst_desc_ids = 7;
  repeated int64 subscribed_regst_desc_ids = 8;
  // for CompTask
  int64 parallel_id = 1000;
}
